pkg/providers/v1/instances.go

```diff
@@ -36,40 +37,29 @@ import (
 // awsInstanceRegMatch represents Regex Match for AWS instance.
 var awsInstanceRegMatch = regexp.MustCompile("^i-[^/]*$")
 
-// InstanceID represents the ID of the instance in the AWS API, e.g. i-12345678
-// The "traditional" format is "i-12345678"
-// A new longer format is also being introduced: "i-12345678abcdef01"
-// We should not assume anything about the length or format, though it seems
-// reasonable to assume that instances will continue to start with "i-".
-type InstanceID string
-
-func (i InstanceID) awsString() *string {
-	return aws.String(string(i))
-}
-
-// KubernetesInstanceID represents the id for an instance in the kubernetes API;
-// the following form
+// ParseProviderID turns a Kubernetes ProviderID into an AWS node id
+// the following are forms of ProviderIDs that are supported:
 //   - aws:///<zone>/<awsInstanceId>
 //   - aws:////<awsInstanceId>
 //   - aws:///<zone>/fargate-<eni-ip-address>
 //   - <awsInstanceId>
-type KubernetesInstanceID string
-
-// MapToAWSInstanceID extracts the InstanceID from the KubernetesInstanceID
-func (name KubernetesInstanceID) MapToAWSInstanceID() (InstanceID, error) {
-	s := string(name)
-
+func ParseProviderID(providerID string) (awsnode.NodeID, error) {
```

Comment:
i observed that we are making call to get variant followed by this method in most places. Can we return the variant also such that the caller can get both at the same time ?